python/doc/examples/calibration/bayesian_calibration/plot_gibbs_simus.py

```diff
@@ -0,0 +1,602 @@
+
+trunc_cond_Y = ot.ComposedDistribution(  marginals_trunc  )
+RV_Y = ot.RandomVector( trunc_cond_Y )
+
+# Step 2: Source link function (SLF), indicating the parameters of the truncated normal
+# as a function of theta and tau.
+# TRICK: use the getParameterDescription() method to know what the SLF needs to output
+print( trunc_cond_Y.getParameterDescription() )
+
+
+# %%
+# Now, use the getParameter() method to extract nominal values of the parameters as a blueprint for the SLF
+
+gen_params = np.array( trunc_cond_Y.getParameter() )
+
+# NOTE: only the mean and std need to be changed, the upper and lower bounds remain fixed:
+
+def pySLF_Y(x):
+    """
+    source link function for Y conditional density
+    
+    Input
+        x: vector with length (p + 1 + n), containing the current state of (theta, tau, Y) 
+    
+    Output
+        params: vector with length 4*n, corresponding to mean, std, a and b, for each component of Y
+        
+    Notes
+        a and b represent the upper and lower bounds for the truncated normals    
+    """
+    theta = [x[i] for i in range(p)]
+    tau = x[p]
+    # compute conditional mean and standard deviates
+    mean = np.dot( X, theta )
+    std = np.sqrt(1. / tau + 1. / Q)
+    # inject values in blueprint 
+    params = gen_params.copy()
+    params[::4] = mean
+    params[1::4] = std.ravel()
+    return params
+
+# %%
+# 2.2.2. Updating :math:`\theta`
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# 
+# :math:`\pi(\theta|Y, \tau, Y^{obs})=\pi(\theta|Y,\tau).` Due to the
+# partial conjugacy of the conditional prior :math:`\pi(\theta|\tau),`
+# this is explicit, and given by the following box-constrained
+# multivariate normal:
+# 
+# .. math::
+# 
+#     \theta|Y,\tau \sim \mathcal N_4( \mu_n ; \Sigma_n ){\bf 1}_{\{\theta_{\min} \leq \theta \leq \theta_{\max}\}},
+# 
+# with
+# 
+# .. math::
+# 
+#     \mu_n = \mu_0 + \Sigma_0X^\top(\mat{X} \Sigma_0 \mat{X}^\top + \mat{I_n} + \tau \mat{Q}^{-1})^{-1}(Y - \mat{X} \mu_0) \\
+#     \Sigma_n = \tau^{-1}\Sigma_0 - \tau^{-1}\Sigma_0 \mat{X}^\top (\mat{X} \Sigma_0 \mat{X}^\top + \mat{I_n} + \tau \mat{Q}^{-1})^{-1} \mat{X} \Sigma_0,
+# 
+# or equivalently, thanks to the matrix Woodsbury identity:
+# 
+# .. math::
+# 
+#     \mu_n =  (\Sigma_0^{-1} + \mat{X}^\top (\mat{I_n} + \tau \mat{Q}^{-1})^{-1} \mat{X})^{-1}(\Sigma_0^{-1} \mu_0 + \mat{X}^\top (\mat{I_n} + \tau \mat{Q}^{-1})^{-1} Y)\\
+#     \Sigma_n = \tau^{-1}(\Sigma_0^{-1} + X^\top (I_n + \tau Q^{-1})^{-1} X)^{-1}.
+# 
+# By having all hyperparameters go to :math:`0,` we obtain the following
+# simplified form:
+# 
+# .. math::
+# 
+#     \mu_n =  (\mat{X}^\top (\mat{I_n} + \tau \mat{Q}^{-1})^{-1} \mat{X})^{-1}(X^\top (I_n + \tau \mat{Q}^{-1})^{-1} Y)\\
+#     \Sigma_n = \tau^{-1}(\mat{X}^\top (\mat{I_n} + \tau \mat{Q}^{-1})^{-1} \mat{X})^{-1}.
+# 
+# Explicit simulation from a box-constrained multivariate normal can be
+# done following the algorithm in [koch2019]_:
+# 
```

Comment:
Have you seen the following comment on arxiv?

"There is an error in the the technical proofs for the proposed algorithms' validity. While the proposed algorithms can in many cases produce approximate draws from the described target distributions, they do not produce exact draws"

Reply:
The article has been retracted from arxiv. IMO the algorithm is wrong and should not be used as an example.

Reply:
Thanks for your feedback Régis! I hadn't noticed the remark, but as the paper presents several algorithms, it may be aimed to a different method. I checked the proof for the box-constrained multivariate normal algorithm, which is quite straightforward, and it seemed ok to me. I will double check and if it is fine, maybe add it as a theory section.